crit/decode.go

```diff
@@ -0,0 +1,270 @@
+package crit
+
+import (
+	"encoding/base64"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+
+	"github.com/checkpoint-restore/go-criu/v5/crit/images"
+	"github.com/checkpoint-restore/go-criu/v5/magic"
+	"google.golang.org/protobuf/proto"
+)
+
+func decodeImg(f *os.File, noPayload bool) (*CriuImage, error) {
+	magicMap := magic.LoadMagic()
+	img := CriuImage{}
+	var err error
+
+	// Read magic
+	magicBuf := make([]byte, 4)
+	if _, err := f.Read(magicBuf); err != nil {
+		return nil, err
+	}
+	magic := uint64(binary.LittleEndian.Uint32(magicBuf))
+	if magic == magicMap.ByName["IMG_COMMON"] ||
+		magic == magicMap.ByName["IMG_SERVICE"] {
+		if _, err := f.Read(magicBuf); err != nil {
+			return nil, err
+		}
+		magic = uint64(binary.LittleEndian.Uint32(magicBuf))
+	}
```

Comment:
The code for reading the magic in `decodeImg()` and `countImg()` looks very similar. Perhaps we could move it into a function (e.g., `readMagic()`) to make it reusable?